notebooks/classification/top_label_calibration.ipynb

```diff
@@ -0,0 +1,413 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Top-label calibration for outputs of ML models\n",
+    "\n",
+    "The goal of this notebook is to present :class:`mapie.calibration.MapieCalibrator` by comparing it to the method presented in the paper for Top-label calibration [1].\n",
+    "\n",
+    "[1] Gupta, Chirag, and Aaditya K. Ramdas. \"Top-label calibration and multiclass-to-binary reductions.\" arXiv preprint arXiv:2107.08353 (2021)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/scikit-learn-contrib/MAPIE/blob/master/notebooks/classification/top_label_calibration.ipynb)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Tutorial preparation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "%matplotlib inline\n",
+    "\n",
+    "from sklearn.datasets import make_classification\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "from sklearn.metrics import classification_report, confusion_matrix, ConfusionMatrixDisplay\n",
+    "from sklearn.ensemble import RandomForestClassifier\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Import calibration GitHub\n",
+    "Import the calibration GitHub of for the paper on Top-label calibration [1].\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Cloning into 'df-posthoc-calibration'...\n",
+      "remote: Enumerating objects: 309, done.\u001b[K\n",
+      "remote: Counting objects: 100% (197/197), done.\u001b[K\n",
+      "remote: Compressing objects: 100% (105/105), done.\u001b[K\n",
+      "remote: Total 309 (delta 101), reused 183 (delta 89), pack-reused 112\u001b[K\n",
+      "Receiving objects: 100% (309/309), 60.76 MiB | 3.74 MiB/s, done.\n",
+      "Resolving deltas: 100% (132/132), done.\n"
+     ]
+    }
+   ],
+   "source": [
+    "!git clone https://github.com/AIgen/df-posthoc-calibration"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sys\n",
```

Comment:
all imports in the same cell

Reply:
done. Only doing the import for the Githubs in different cells. Is that okay?

Reply:
do the gitclone in the first cell of your notebook, and then import everything in the same cell. Juste be carefull, is there is a push on the master branch on the repo, your notebook might not work anymore, chck if it is possible to specify a commit when cloning

Reply:
I would agree. It would seem like that will be difficult, because in theory, I would have to do a `git checkout <commit>` However it's not finding it's looking for a commit in MAPIE and not that repository.

Reply:
Let's discuss this face to face